subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java

```diff
@@ -182,15 +189,32 @@ public void registerBuildSessionServices(ServiceRegistration registration) {
         registration.addProvider(new BuildSessionServices());
     }
 
+    private static class GlobalScopeServices {
+        FileHasherStatistics.Collector createCachingFileHasherStatisticsCollector() {
+            return new FileHasherStatistics.Collector();
+        }
+
+        DirectorySnapshotterStatistics.Collector createDirectorySnapshotterStatisticsCollector() {
+            return new DirectorySnapshotterStatistics.Collector();
+        }
+    }
+
     @VisibleForTesting
     static class GradleUserHomeServices {
 
         CrossBuildFileHashCache createCrossBuildFileHashCache(CacheRepository cacheRepository, InMemoryCacheDecoratorFactory inMemoryCacheDecoratorFactory) {
             return new CrossBuildFileHashCache(null, cacheRepository, inMemoryCacheDecoratorFactory, CrossBuildFileHashCache.Kind.FILE_HASHES);
         }
 
-        FileHasher createCachingFileHasher(StringInterner stringInterner, CrossBuildFileHashCache fileStore, FileSystem fileSystem, GradleUserHomeScopeFileTimeStampInspector fileTimeStampInspector, StreamHasher streamHasher) {
-            CachingFileHasher fileHasher = new CachingFileHasher(new DefaultFileHasher(streamHasher), fileStore, stringInterner, fileTimeStampInspector, "fileHashes", fileSystem);
+        FileHasher createCachingFileHasher(
+            FileHasherStatistics.Collector statisticsCollector,
+            CrossBuildFileHashCache fileStore,
+            FileSystem fileSystem,
+            GradleUserHomeScopeFileTimeStampInspector fileTimeStampInspector,
+            StreamHasher streamHasher,
+            StringInterner stringInterner
+        ) {
+            CachingFileHasher fileHasher = new CachingFileHasher(new DefaultFileHasher(streamHasher), fileStore, stringInterner, fileTimeStampInspector, "fileHashes", fileSystem, 400000, statisticsCollector);
```

Comment:
Can you extract `400000` into a constant with some description?

Reply:
I don't see a point in doing so. I don't have any context about that number, just moved it around a bit. IntelliJ gives you the purpose of it (which is pretty obscure as it is):

![image](https://user-images.githubusercontent.com/495366/95986451-4db22a00-0e26-11eb-972c-6ddd4f21f122.png)

Reply:
The number now is twice in `VirtualFileSystemServices` and was only once in the constructor you inlined. If we change it, I suppose it should be changed consistently in both places or at least made a conscious decision to only change one number. When there is no constant, I think one can easily forget/overlook the other number.

Reply:
👍 Fixed.